crates/sui-replay/src/lib.rs

```diff
@@ -531,3 +534,18 @@ pub async fn execute_replay_command(
         }
     })
 }
+
+pub(crate) fn chain_from_chain_id(chain: &str) -> Chain {
+    let mainnet_chain_id = format!("{}", get_mainnet_chain_identifier());
+    // TODO: Since testnet periodically resets, we need to ensure that the chain id
+    // is updated to the latest one.
+    let testnet_chain_id = format!("{}", get_testnet_chain_identifier());
+
+    if mainnet_chain_id == chain {
+        Chain::Mainnet
+    } else if testnet_chain_id == chain {
+        Chain::Testnet
+    } else {
+        Chain::Unknown
+    }
+}
```

Comment:
Still not quite sure why we can't use https://github.com/MystenLabs/sui/blob/c5ae3de27d17f6ae9b7f875f7b3e6966320dc706/crates/sui-types/src/digests.rs#L162-L172, which would essentially replace this entire function.

Reply:
I guess this takes a string rather than the typed `ChainIdentifier`.

Reply:
Addressed offline